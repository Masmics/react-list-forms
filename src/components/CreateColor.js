import React, { PureComponent } from 'react';
import { render } from 'react-dom';
import PropTypes from 'prop-types';

// create a CreateColor component takes an addColor property
// CreateColor is a form with a name and color input
// on submitting the form invoke the addColor property to add a new color

export default class CreateColor extends PureComponent {

  static propTypes = {
    addColor: PropTypes.func.isRequired
  };

  }
  state = {
    name: '',
    color: '#000000'
  }  

  handleSubmit = event => {
    event.preventDefault();
    console.log(this.state);
    const { name, color } = this.state;

    this.props.addColor({ name, hex: color });
  }

  handleColorNameChange = ({ target }) => {
    this.setState({ name: target.value });
  }

  handleColorChange = ({ target }) => {
    this.setState({ color: target.value });
  }

  render() {
    const { name, color } = this.state;
    return (
      <form onSubmit={this.handleSubmit}>
        <input name="name" type="text" value={name} onChange={this.handleColorNameChange} />
        <input color="color" type="color" value={color} onChange={this.handleColorChange} />
        <button>Add Color</button>
      </form>
    )
  }
}
