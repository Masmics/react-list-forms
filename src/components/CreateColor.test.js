import React from 'react';
import { shallow } from 'enzyme';
import CreateColor from './CreateColor';

describe('CreateColor component', () => {
  it('creates a new color', () => {
    const wrapper = shallow(<CreateColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
