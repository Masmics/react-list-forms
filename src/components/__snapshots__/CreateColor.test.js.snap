// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateColor component creates a new color 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Enter Color Name:
  </p>
  <input
    name="name"
    onChange={[Function]}
    type="text"
    value=""
  />
  <input
    name="color"
    onChange={[Function]}
    type="color"
    value="#000000"
  />
  <p>
    Select Color:
  </p>
  <button>
    Add Color
  </button>
</form>
`;
